File: java/PraUTS/src/main/java/PraUTS/Android.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package PraUTS;

import java.util.Date;

/**
 *
 * @author dodiaditya
 */
public class Android extends OperatingSystem {
    private String merk;
    
    public Android(String version, Date release_date, String language, String merk) {
        this(version, release_date, language, merk, null);
    }
    
    public Android(String version, Date release_date, String language, String merk, String serial_number) {
        super("Android", version, release_date, "Google", language, serial_number);
        this.merk = merk;
    }
    
    @Override
    public void printAdditionalInfo() {
        System.out.println("Merk            : " + merk);
    }
    
    public String getMerk() {
        return this.merk;
    }
    
    public void openGoogleAssistant() {
        System.out.println("OK Google!");
    }
}

File: swift/PraUTS/PraUTS/Android.swift
//
//  Android.swift
//  PraUTS
//
//  Created by Dodi Aditya on 06/11/23.
//

import Foundation

class Android: OperatingSystem {
    private(set) var merk: String
    
    init(version: String, releaseDate: Date, language: String, merk: String, serialNumber: String) {
        self.merk = merk
        super.init(name: "Android", version: version, releaseDate: releaseDate, developer: "Google", language: language, serialNumber: serialNumber)
    }
    
    convenience init(version: String, releaseDate: Date, developer: String, language: String, merk: String) {
        self.init(version: version, releaseDate: releaseDate, language: language, merk: merk, serialNumber: "")
    }
    
    override func printInfo() {
        super.printInfo()
        print("Merk            : \(merk)" )
    }
    
    func openGoogleAssistant() {
        print("OK Google!")
    }
}
